sale_loyalty_partner_applicability/tests/test_sale_loyalty_partner_applicability.py

```diff
@@ -0,0 +1,233 @@
+                        },
+                    ),
+                ],
+                "reward_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "reward_type": "discount",
+                            "required_points": 1,
+                            "discount": 10,
+                            "discount_mode": "percent",
+                            "discount_applicability": "order",
+                        },
+                    )
+                ],
+            }
+        )
+        cls.next_order_coupon = cls.env["loyalty.program"].create(
+            {
+                "name": "Test Loyalty Limit Next Order Coupons",
+                "program_type": "next_order_coupons",
+                "trigger": "auto",
+                "applies_on": "future",
+                "rule_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "rule_partners_domain": [("id", "=", cls.partner1.id)],
+                            "reward_point_mode": "order",
+                            "minimum_qty": 1,
+                        },
+                    ),
+                ],
+                "reward_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "reward_type": "discount",
+                            "required_points": 1,
+                            "discount": 10,
+                            "discount_mode": "percent",
+                            "discount_applicability": "order",
+                        },
+                    )
+                ],
+            }
+        )
+
+    def _create_sale(self, partner):
+        sale_form = Form(self.env["sale.order"])
+        sale_form.partner_id = partner
+        with sale_form.order_line.new() as line_form:
+            line_form.product_id = self.product_a
+            line_form.product_uom_qty = 1
+        with sale_form.order_line.new() as line_form:
+            line_form.product_id = self.product_b
+            line_form.product_uom_qty = 1
+        return sale_form.save()
+
+    def _action_apply_program(self, sale, program):
+        sale._update_programs_and_rewards()
+        wizard = (
+            self.env["sale.loyalty.reward.wizard"]
+            .with_context(active_id=sale)
+            .create({"selected_reward_id": program.reward_ids.id})
+        )
+        wizard.action_apply()
+
+    def _apply_promo_code(self, order, code, no_reward_fail=True):
+        status = order._try_apply_code(code)
+        if "error" in status:
+            raise ValidationError(status["error"])
+        if not status and no_reward_fail:
+            # Can happen if global discount got filtered out in `_get_claimable_rewards`
+            raise ValidationError(_("No reward to claim with this coupon"))
```

Comment:
Is this condition ever tested?

Reply:
In the case of the tests of this module it is only necessary to check if there is an error when applying the code, therefore I have removed the conditions that are not necessary, thank thanks.